lib/wallet/src/Cardano/Wallet/DB/Sqlite/Types.hs

```diff
@@ -753,3 +759,38 @@ iso8601DateFormatHMS :: String
 -- Equivalent to `iso8601DateFormatHMS (Just "%H:%M:%S")`
 -- The function `iso8601DateFormatHMS` has been deprecated from the `time` library.
 iso8601DateFormatHMS = "%Y-%m-%d %H:%M:%S"
+
+instance PersistField Slot where
+    toPersistValue Origin = toPersistValue ((-1) :: Int64)
+    toPersistValue (At s) = toPersistValue (fromIntegral $ unSlotNo s :: Int64)
+    fromPersistValue v = case fromPersistValue v of
+        Right (PersistInt64 x)
+            | x < (-1) -> Left "Slot must be positive or -1"
+            | x == (-1) -> Right Origin
+            | otherwise -> Right . At . SlotNo . fromIntegral $ x
+        _ -> Left "Slot must be an Int64"
+
+instance PersistFieldSql Slot where
+    sqlType _ = sqlType (Proxy @Int64)
+
+instance PersistField Pruned where
+    toPersistValue NotPruned = toPersistValue @(Maybe Word64) Nothing
+    toPersistValue (PrunedUpTo h) = toPersistValue (Just h)
+    fromPersistValue v = case fromPersistValue v of
+        Right (PersistNull) -> Right NotPruned
+        Right (PersistInt64 x) -> Right $ PrunedUpTo $ fromIntegral x
+        _ -> Left "Pruned must be an Int64 or Null"
+
+instance PersistFieldSql Pruned where
+    sqlType _ = sqlType (Proxy @(Maybe Word64))
+
+instance PersistField Spent where
+    toPersistValue Unspent = toPersistValue @(Maybe Word64) Nothing
+    toPersistValue (Spent h) = toPersistValue (Just h)
+    fromPersistValue v = case fromPersistValue v of
+        Right (PersistNull) -> Right Unspent
+        Right (PersistInt64 x) -> Right $ Spent $ fromIntegral x
+        _ -> Left "Spent must be an Int64 or Null"
+
+instance PersistFieldSql Spent where
+    sqlType _ = sqlType (Proxy @(Maybe Word64))
```

Comment:
```suggestion
instance PersistField Spent where
    toPersistValue Unspent = toPersistValue @(Maybe Word64) Nothing
    toPersistValue (Spent h) = toPersistValue (Just h)
    fromPersistValue v = case fromPersistValue @(Maybe Word64) v of
        Right Nothing -> Right Unspent
        Right (Just x) -> Right $ Spent $ fromIntegral x
        Left _ -> Left "Spent must be a Word64 or Null"

instance PersistFieldSql Spent where
    sqlType _ = sqlType (Proxy @(Maybe Word64))
```

I'm not entirely sure if `Word64` is really stored as `Int64` — that format would may be missing a single bit… 🤔